bot/seasons/evergreen/trivia_quiz.py

```diff
@@ -28,149 +26,215 @@ class TriviaQuiz(commands.Cog):
     def __init__(self, bot: commands.Bot) -> None:
         self.bot = bot
         self.questions = self.load_questions()
-        self.game_status = {}
-        self.game_owners = {}
+        self.game_status = {}  # A variable to store the game status: either running or not running.
+        self.game_owners = {}  # A variable to store the person's ID who started the quiz game in a channel.
         self.question_limit = 4
-        self.player_dict = {}
+        self.player_scores = {}  # A variable to store all player's scores for a bot session.
+        self.game_player_scores = {}  # A variable to store temporary game player's scores.
         self.categories = {
             "general": "Test your general knowledge"
             # "retro": "Questions related to retro gaming."
         }
 
     @staticmethod
     def load_questions() -> dict:
-        """Load the questions from  json file."""
+        """Load the questions from the JSON file."""
         p = Path("bot", "resources", "evergreen", "trivia_quiz.json")
         with p.open() as json_data:
             questions = json.load(json_data)
             return questions
 
-    @commands.command(name="quiz", aliases=["trivia"])
-    async def quiz_game(self, ctx: commands.Context, category: str = "general") -> None:
+    @commands.group(name="quiz", aliases=["trivia"], invoke_without_command=True)
+    async def quiz_game(self, ctx: commands.Context, category: str = None) -> None:
         """
-        Start/Stop a quiz!
-
-        arguments:
-        option:
-        - start : to start a quiz in a channel
-        - stop  : stop the quiz running in that channel.
+        Start a quiz!
 
         Questions for the quiz can be selected from the following categories:
         - general : Test your general knowledge. (default)
-        (we wil be adding more later)
+        (More to come!)
         """
-        category = category.lower()
-
         if ctx.channel.id not in self.game_status:
             self.game_status[ctx.channel.id] = False
-            self.player_dict[ctx.channel.id] = {}
 
-        if not self.game_status[ctx.channel.id]:
+        if ctx.channel.id not in self.game_player_scores:
+            self.game_player_scores[ctx.channel.id] = {}
+
+        # Stop game if running.
+        if self.game_status[ctx.channel.id] is True:
+            return await ctx.send(
+                f"Game is already running..."
+                f"do `{self.bot.command_prefix}quiz stop`"
+            )
+
+        # Send embed showing available categori es if inputted category is invalid.
+        if category is None:
+            category = random.choice(list(self.categories))
+        if category.lower() not in self.categories:
```

Comment:
```suggestion

        category = category.lower()
        if category not in self.categories:
```

Reply:
This wont work cuz if category is None, then it will throw error

Reply:
The category will never be none because that is checked right above this line.

Reply:
nvm  i get it, its coming after the 1st if , so i guess it should be fine.

Reply:
what difference will it make btw?

Reply:
> Fix KeyError if quiz category is not lowercase